serial_posix.go

```diff
@@ -35,27 +34,92 @@ func openPort(name string, baud int, readTimeout time.Duration) (p *Port, err er
 		f.Close()
 		return nil, err
 	}
-	var speed C.speed_t
-	switch baud {
-	case 115200:
-		speed = C.B115200
-	case 57600:
-		speed = C.B57600
-	case 38400:
-		speed = C.B38400
-	case 19200:
-		speed = C.B19200
-	case 9600:
-		speed = C.B9600
-	case 4800:
-		speed = C.B4800
-	case 2400:
-		speed = C.B2400
-	default:
-		f.Close()
-		return nil, fmt.Errorf("Unknown baud rate %v", baud)
+
+  var bauds = map[int]C.speed_t{
+		50:      C.B50,
+		75:      C.B75,
+		110:     C.B110,
+		134:     C.B134,
+		150:     C.B150,
+		200:     C.B200,
+		300:     C.B300,
+		600:     C.B600,
+		1200:    C.B1200,
+		1800:    C.B1800,
+		2400:    C.B2400,
+		4800:    C.B4800,
+		9600:    C.B9600,
+		19200:   C.B19200,
+		38400:   C.B38400,
+		57600:   C.B57600,
+		115200:  C.B115200,
+		230400:  C.B230400,
+		460800:  C.B460800,
+		500000:  C.B500000,
+		576000:  C.B576000,
+		921600:  C.B921600,
+		1000000: C.B1000000,
+		1152000: C.B1152000,
+		1500000: C.B1500000,
```

Comment:
230400 works on OSX -- only added the higher rates 2 days ago (from your serial_linux implementation), and don't have an OSX test machine at the minute. Your right - this area is pretty wooley. A note on cross-platform portability in the GLIBC manual lists the constants up to B430600 as being standard
(http://www.gnu.org/software/libc/manual/html_node/Line-Speed.html), but a bsd version of termios.h (which is probably a closer match to OSX than the linux version) only goes up to 230400
(http://fxr.watson.org/fxr/source/sys/termios.h?v=OPENBSD)

So could end the list at 230400 if you're happy with that, and review at a later date if I get access to an OSX machine to test higher baud rates on?